File: python worker invokation/request_json_adapter.py
import json

class RequestJsonAdapter():
    """This Class provides static methods to convert function calls to json"""
    def __init__(self) -> None:
        pass

    @staticmethod
    def get_update_request(sensor_type,value):
        "update a sensor value"
        res ={
            "type":"update_request",
            "sensor_type":str(sensor_type),
            "sensor_value":str(value)
        }
        return json.dumps(res)

    @staticmethod
    def get_sensor_response(sensor_type, value):
        "get the value of a sensor"
        res = {
            "type":"sensor_response",
            "sensor_type": str(sensor_type),
            "value": str(value)
        }
        return json.dumps(res)

    @staticmethod
    def get_sensor_request(sensor_type):
        "get the value of a sensor"
        res = {
            "type":"sensor_request",
            "sensor_type": str(sensor_type)
        }
        return json.dumps(res)

    

File: Apps/SensorTryOut/app/src/main/java/com/example/sensortryout/RequestJsonAdapter.java
package com.example.sensortryout;

import org.json.JSONException;
import org.json.JSONObject;

public class RequestJsonAdapter {

    public static JSONObject get_update_request(String sensorType, String sensorValue) throws JSONException {
        JSONObject jo = new JSONObject();
        jo.put("type", "update_request");
        jo.put("sensor_type",sensorType);
        jo.put("sensor_value", sensorValue);
        return jo;
    }

    public static JSONObject get_rpc_response(String command, String value) throws JSONException {
        JSONObject jo = new JSONObject();
        jo.put("type", "rpc_response");
        jo.put("command",command);
        jo.put("value", value);
        return jo;
    }


}
